pkg/controller/apiextensions/composite/composed/api.go

```diff
@@ -164,3 +167,36 @@ func (cdf *APIConnectionDetailsFetcher) Fetch(ctx context.Context, cd resource.C
 
 	return conn, nil
 }
+
+// DefaultReadinessProber is a readiness prober which returns whether the composed
+// resource is ready or not.
+type DefaultReadinessProber struct{}
```

Comment:
Nit: Could we use a `ReadinessProberFn`, rather than using an empty struct named `Default<T>` to satisfy our interface? It doesn't seem like this needs to be a struct.

Edit: Ah, I see this is an established convention in this file. It's not my preferred convention, but I'm happy to stick to it.

Reply:
That would've been nicer indeed. I want to defer it to later though.